### PR TITLE
Page banner standards

### PR DESCRIPTION
**Pull request by ineilson**
Applies coding standards to files changed by #403

### CODE REVIEW

File: htdocs/web_portal/components/Draw_Components/draw_page_components.php [hunk 1]
```diff
@@ -18,162 +18,184 @@
      * See the License for the specific language governing permissions and
      * limitations under the License.
      *
-     *====================================================== */
+     * PHPMD does not parse use of $properties with :: scope resolution
+     * @SuppressWarnings(PHPMD.UndefinedVariable)
+     */
 
     /* Draws the first part of the page (headers, left hand menu) */
```
Comment (gregcorbett) [inline htdocs/web_portal/components/Draw_Components/draw_page_components.php:25]: Why not unindent the comments as well?

File: lib/Gocdb_Services/Config.php [hunk 2]
```diff
@@ -19,26 +21,33 @@
  *
  * @author David Meredith <david.meredith@stfc.ac.uk>
  * @author John Casson
+ *
+ * @SuppressWarnings(PHPMD.CamelCaseMethodName)
+ * @phpcs:disable PSR1.Methods.CamelCapsMethodName.NotCamelCaps
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Config.php:26]: what is the effect of this?

File: lib/Gocdb_Services/Config.php [hunk 3]
```diff
@@ -94,88 +105,103 @@ public function setLocalInfoFileLocation($filePath){
...
-            throw new \ErrorException('Failed to find local_info section without url in configuration file '.$path);
+        $unqualified = $base->xpath("//local_info[not(@url)]");
+        if (!$unqualified) {
+            throw new \ErrorException('Failed to find local_info section without url in configuration file ' . $path);
         }
 
         if (count($unqualified) != 1) {
-            throw new \ErrorException('Only one local_info element without url attribute is allowed in configuration file '.$path);
+            throw new \ErrorException(
+                'Only one local_info element without url attribute is allowed in configuration file ' .
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Config.php:170]: Given we are already concatinating here, can this long line be split up?

File: lib/Gocdb_Services/Config.php [hunk 4]
```diff
@@ -94,88 +105,103 @@ public function setLocalInfoFileLocation($filePath){
...
 
         if (!is_null($url)) {
             // Find any elements matching the given url
             $qualified = $base->xpath("//local_info[@url=\"$url\"]");
 
             if (count($qualified) != 0) {
                 if (count($qualified) != 1) {
-                    throw new \ErrorException('Duplicate local_info elements with same url attribute found in configuration file '.$path);
+                    throw new \ErrorException(
+                        'Duplicate local_info elements with same url attribute found in configuration file ' .
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Config.php:184]: Given we are already concatinating here, can this long line be split up?

File: lib/Gocdb_Services/Config.php [hunk 5]
```diff
@@ -185,30 +211,36 @@ private function throwXmlErrors ($message) {
...
+                $elemPath = implode("/", $keys);
+                $elem = $defaultInfo->xpath($elemPath);
+                if (!$elem) {
+                    throw new \ErrorException("Did not find elements $elemPath in input configuration file.");
                 }
                 if (count($elem)) {
                     if (count($elem) != 1) {
-                        throw new \ErrorException('Duplicate input configuration file element specifications ('.count($elem).') for "'.implode('/',$keys).'"');
+                        throw new \ErrorException(
+                            'Duplicate input configuration file element specifications (' . count($elem) .
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Config.php:231]: Given we are already concatinating here, can this long line be split up?

File: lib/Gocdb_Services/Config.php [hunk 6]
```diff
@@ -185,30 +211,36 @@ private function throwXmlErrors ($message) {
...
+                            ') for "' . implode('/', $keys) . '"'
+                        );
                     }
                     # ???? How else to force a self-reference rather than override the array value ????
                     $elem[0][0] = (string)$iterator->current();
                 } else {
                     // Do we want to create it here ??
-                    throw new \ErrorException("Input configuration file override element $p not found in default section: ".$iterator->key().' => '.$iterator->current());
+                    throw new \ErrorException(
+                        "Input configuration file override element $elemPath not found in default section: " .
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Config.php:240]: Given we are already concatinating here, can this long line be split up?

File: lib/Gocdb_Services/Config.php (hunk 2 above)
Comment (ineilson) [reply lib/Gocdb_Services/Config.php:26]: Suppressing
image
which would need wider changes.

File: lib/Gocdb_Services/Config.php (hunk 3 above)
Comment (ineilson) [reply lib/Gocdb_Services/Config.php:170]: It could be, but seems not overlong to me. Standard is set to max of 120 chars. What are you suggesting?

Comment (gregcorbett) [reply lib/Gocdb_Services/Config.php:170]: it was long enough that it wrapped on GitHub for me 😢 

image

Could go to <80? PSr12 does say lines SHOULD NOT be longer than that.

Comment (gregcorbett) [reply lib/Gocdb_Services/Config.php:170]: i.e. 

```suggestion
                'Only one local_info element without url attribute is ' . 
                'allowed in configuration file ' .
```

File: htdocs/web_portal/components/Draw_Components/draw_page_components.php (hunk 1 above)
Comment (ineilson) [reply htdocs/web_portal/components/Draw_Components/draw_page_components.php:25]: No reason. The auto-reformat didn't touch them, but I'll update.

File: lib/Gocdb_Services/Config.php (hunk 3 above)
Comment (ineilson) [reply lib/Gocdb_Services/Config.php:170]: OK. I'm not a fan of 80 chars, being an arbitrary legacy, but I can see the point and it works here for concatenations.

**Review by gregcorbett — APPROVED** [review]

**Comment by gregcorbett** [comment]
rebasing to resolve "conflicts" caused by [62d3fc5](https://github.com/GOCDB/gocdb/pull/404/commits/62d3fc518af6fd92b04843bbd4c63d675e9a0822)

**Pull request merged**